fendermint/app/src/main.rs

```diff
@@ -30,6 +32,58 @@ async fn main() {
     }
 }
 
+fn create_log(level: tracing::Level, log_dir: &Path) -> anyhow::Result<()> {
+    let log_folder = expand_tilde(log_dir)
+        .to_str()
+        .ok_or_else(|| anyhow!("cannot parse log folder"))?
+        .to_string();
+    std::fs::create_dir_all(&log_folder).context("cannot create log folder")?;
+
+    let console_appender = config::Appender::Console;
+    let topdown_appender = config::Appender::RollingFile {
+        path: format!("{log_folder}/topdown.log"),
+        policy: Policy {
+            maximum_file_size: "10mb".to_string(),
+            // we keep the last 5 log files, older files will be deleted
+            max_size_roll_backups: 5,
+            pattern: None,
+        },
+    };
+
+    // debug level logger
+    let topdown_logger = config::Logger {
+        level: config::LevelFilter::DEBUG,
+        appenders: literally::hset! { AppenderId::from("topdown") },
+        format: Format::default(),
+    };
+
+    // default logging output info log to console
+    let default = config::Logger {
+        level: config::LevelFilter::from_str(level.as_str())?,
+        appenders: literally::hset! { AppenderId::from("console") },
+        format: Format::default(),
+    };
+
+    let config = Config {
+        default,
+        loggers: literally::hmap! {
+            Target::from("fendermint_vm_topdown") => topdown_logger.clone(),
+            Target::from("fendermint_vm_interpreter::chain") => topdown_logger,
```

Comment:
Nice that we can specify the sources! Buty why is it restricted to topdown, why not everything under the fendermint packages? At least we wouldn't get the debug logs from 3rd party libraries, but see everything we produce, and topdown might not be the only interesting thing to look at.

Reply:
yeap, I made it Optional as well. If not set, just log to console.